Neos.Flow/Classes/Security/Authentication/Token/UsernamePassword.php

```diff
@@ -56,6 +78,18 @@ public function updateCredentials(ActionRequest $actionRequest)
         }
     }
 
+    public function getUsername(): string
+    {
+        $arguments = array_merge($this->actionRequest->getArguments(), $this->actionRequest->getInternalArguments());
+        return ObjectAccess::getPropertyPath($arguments, $this->options['usernamePostField'] ?? self::DEFAULT_USERNAME_POST_FIELD);
+    }
+
+    public function getPassword(): string
+    {
+        $arguments = array_merge($this->actionRequest->getArguments(), $this->actionRequest->getInternalArguments());
```

Comment:
see above

Reply:
also: merging all arguments was my hack to mix internal and regular args, but we should not do it twice :)

Reply:
why would we want to merge here or in general?

Reply:
because of the weird way Flow separates internal and default arguments.
```php
ObjectAccess::getPropertyPath($this->actionRequest->getArguments(), '__authentication.Neos.Flow.Security.Authentication.Token.UsernamePassword.username');
```
does not work